File: javascript/core/src/verbs/fill.ts
/*!
 * Copyright (c) Microsoft. All rights reserved.
 * Licensed under the MIT license. See LICENSE file in the project.
 */

import type { FillArgs } from '@datashaper/schema'

import type { ColumnTableStep } from './util/factories.js'
import { stepVerbFactory } from './util/factories.js'

export const fillStep: ColumnTableStep<FillArgs> = (input, { value, to }) => {
	const fn = (_d: any, $: any) => $.value
	return input.params({ value }).derive({ [to]: fn })
}

export const fill = stepVerbFactory(fillStep)

File: python/datashaper/datashaper/engine/verbs/fill.py
#
# Copyright (c) Microsoft. All rights reserved.
# Licensed under the MIT license. See LICENSE file in the project.
#
"""Fill verb implementation."""

from typing import Any, cast

import pandas as pd

from datashaper.engine.create_verb_result import create_verb_result
from datashaper.engine.types import VerbResult
from datashaper.engine.verbs.verb_input import VerbInput
from datashaper.engine.verbs.verbs_mapping import verb
from datashaper.verbs import fill


@verb(name="fill")
def fill_verb(input: VerbInput, **kwargs: Any) -> VerbResult:
    """Fill verb implementation."""
    result = fill(cast(pd.DataFrame, input.get_input()), **kwargs)
    return create_verb_result(result)
